### PR TITLE
Move TaintToleration predicate to its plugin

### LINKED ISSUE #86978: Move tainttoleration predicate logic to its Filter plugin [Migration Phase 2]
/assign @ahg-g 
/sig scheduling
/priority important-soon

Part of #85822

### PR DESCRIPTION
**Pull request by ahg-g**
**What type of PR is this?**
/kind cleanup

**Which issue(s) this PR fixes**:
Fixes #86978

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/cc @alculquicondor

### CODE REVIEW

**Comment by ahg-g** [comment]
@alculquicondor this is ready

/hold cancel

File: pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go [hunk 1]
```diff
@@ -142,8 +156,10 @@ func (pl *TaintToleration) Score(ctx context.Context, state *framework.CycleStat
 func (pl *TaintToleration) NormalizeScore(ctx context.Context, _ *framework.CycleState, pod *v1.Pod, scores framework.NodeScoreList) *framework.Status {
 	// Note that ComputeTaintTolerationPriorityReduce doesn't use priority metadata, hence passing nil here.
 	normalizeFun := priorities.NormalizeReduce(framework.MaxNodeScore, true)
```
Comment (draveness) [inline pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go]: We could use DefaultNormalizeScore in helper package

**Review by draveness — CHANGES_REQUESTED** [review]

File: pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go]: done.

**Review by draveness — APPROVED** [review]
/lgtm

**Comment by alculquicondor** [comment]
@draveness, for future scenarios, prefer to hold for other involved reviewers before LGTMing. Specially for an auto-approved PRs such as this one.
Unless the PR is urgent. But no worries for now, if there is anything, we can do a followup PR.

File: pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go [hunk 2]
```diff
@@ -53,9 +53,23 @@ func (pl *TaintToleration) Name() string {
 
 // Filter invoked at the filter extension point.
 func (pl *TaintToleration) Filter(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeInfo *nodeinfo.NodeInfo) *framework.Status {
-	// Note that PodToleratesNodeTaints doesn't use predicate metadata, hence passing nil here.
-	_, reasons, err := predicates.PodToleratesNodeTaints(pod, nil, nodeInfo)
-	return migration.PredicateResultToFrameworkStatus(reasons, err)
+	if nodeInfo == nil || nodeInfo.Node() == nil {
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go]: do you think we should be getting rid of these now?

File: pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go [hunk 3]
```diff
@@ -23,9 +23,7 @@ import (
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
-	"k8s.io/kubernetes/pkg/scheduler/algorithm/predicates"
-	"k8s.io/kubernetes/pkg/scheduler/algorithm/priorities"
-	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/migration"
+	pluginhelper "k8s.io/kubernetes/pkg/scheduler/framework/plugins/helper"
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go]: super nit: `pluginshelper`

**Comment by draveness** [comment]
> @draveness, for future scenarios, prefer to hold for other involved reviewers before LGTMing. Specially for an auto-approved PRs such as this one.
> Unless the PR is urgent. But no worries for now, if there is anything, we can do a followup PR.

Sure, but I didn't see your response and didn't know whether you are going to review this.

**Comment by alculquicondor** [comment]
Be mindful that people are in different timezones.

**Pull request merged**